addons/web/static/src/js/core/owl_popover.js

```diff
@@ -0,0 +1,272 @@
+odoo.define('web.OwlPopover', function() {
+    'use strict';
+
+    const { Component, hooks, misc, QWeb } = owl;
+    const { Portal } = misc;
+    const { useExternalListener, useRef, useState } = hooks;
+
+    /**
+     * Popover (owl version)
+     *
+     * Represents a bootstrap-style popover handled with pure JS. The popover
+     * will be visually bound to its `target` using an arrow-like '::before'
+     * CSS pseudo-element.
+     * @extends Component
+     **/
+    class Popover extends Component {
+        /**
+         * @param {Object} [props]
+         * @param {string} [props.position='bottom'] 'top', 'bottom', 'left' or 'right'
+         * @param {HTMLElement} [props.target='body'] the target element (the popover's
+         *                      arrow decoration will point to it)
+         * @param {string} [props.title]
+         */
+        constructor() {
+            super(...arguments);
+            this.popoverRef = useRef('popover');
+            this.orderedPositions = ['top', 'right', 'bottom', 'left'];
+            this.requestedPosition = null;
+            this.state = useState({
+                displayed: false,
+            })
+            // without event capture, scroll listener on children nodes will prevent
+            // this handler from being fired
+            useExternalListener(document, 'scroll', this.show, {
+                capture: true,
+            });
+            useExternalListener(window, 'resize', this.show);
```

Comment:
shouldn't those handlers be debounced?

Reply:
also, we could only bind those handlers when there is an open popover, like the click handler, right?

Reply:
yes and yes

Reply:
not sure about the second part

my understanding of `useExternalListener` is that it will only bind/unbind the listener when the component if (un)mounted; i see no reason to do that logic differently for those methods. IMHO it's better to stick with the framework's way of doing things if it supports the use case...

Reply:
Indeed, you can't use useExternalListener if you want to bind the handlers only when the popover is open. In that case, you'll need to 'manually' bind it like you did for click.

Using 'useExternalListener' indeed unbind the handler when the component is unmounted, but in our case we can go further (having the popover in the DOM doesn't mean it is open).

Reply:
and actually, i will debounce the resize listener but i think i'll throttle the scroll listener instead; it provides a smoother UX

Reply:
:+1:

Reply:
you're right, i forgot to keep in my brain that these listener were only needed when it was opened; push done